python/kserve/kserve/protocol/rest/v2_endpoints.py

```diff
@@ -149,12 +149,19 @@ async def infer(
                                                                 request=infer_request,
                                                                 headers=request_headers)
 
-        response, response_headers = self.dataplane.encode(model_name=model_name,
+        response, res_headers = self.dataplane.encode(model_name=model_name,
                                                            response=response,
                                                            headers=response_headers, req_attributes={})
 
+        response_headers.update(res_headers)
+        if 'content-length' in response_headers:
+            del response_headers['content-length']
+        if 'content-type' in response_headers:
+            del response_headers['content-type']
```

Comment:
We need to define a list of the headers that need to propagate back instead of excluding as there are other headers like authorization token which shouldn't be in the response.